src/search/search-index/add.js

```diff
@@ -180,3 +180,31 @@ async function performIndexing(indexDoc) {
         console.error(err)
     }
 }
+
+async function indexingMetaConcurrent(pageId, timestamp) {
+    const reverseIndexDoc = await singleLookup(pageId)
+    reverseIndexDoc.visits.add(timestamp)
+    await index.put(pageId, {
+        ...reverseIndexDoc,
+    })
+}
+
+/**
+ * Adds a new meta index (bookmark or visit) entry.
+ * NOTE: Assumes the existence of indexed `pageId`.
+ *
+ * @param {'bookmark'|'visit'} type Type of meta event.
+ * @param {string} timestamp Timestamp of meta event.
+ * @param {string} pageId ID of page doc to associate with.
+ */
+export const addMetaConcurrent = (type, timestamp, pageId) =>
+    new Promise((resolve, reject) => {
+        indexingMetaConcurrent(pageId, timestamp)
```

Comment:
This call should be inside the `indexQueue.push` callback, before the current `index.put` call, or else it will just be called as soon as this function gets invoked (which is unsafe concurrency-wise, as other running code could be accessing the same key)